utilities/query.py

```diff
@@ -186,16 +208,42 @@ def create_query(user_query, click_prior_query, filters, sort="_score", sortDir=
     return query_obj
 
 
-def search(client, user_query, index="bbuy_products", sort="_score", sortDir="desc"):
+def search(client, user_query, index="bbuy_products", sort="_score", sortDir="desc", vector_query=False):
     #### W3: classify the query
     #### W3: create filters and boosts
     # Note: you may also want to modify the `create_query` method above
-    query_obj = create_query(user_query, click_prior_query=None, filters=None, sort=sort, sortDir=sortDir, source=["name", "shortDescription"])
-    logging.info(query_obj)
-    response = client.search(query_obj, index=index)
-    if response and response['hits']['hits'] and len(response['hits']['hits']) > 0:
-        hits = response['hits']['hits']
+    if vector_query:
+        embedding = bert.encode([user_query])
+        target_size = 10
+        filters = {
+            "bool": {
+                "onSale": True
+            }
+        }
+        size = target_size
+        response_size = 0
+        while True:
+            query_obj = create_vector_query(embedding, filters=filters, size=size)
+            logging.info(query_obj)
+            response = client.search(query_obj, index=index)
+            if response and response['hits']['hits'] and len(response['hits']['hits']) > response_size:
```

Comment:
any reason why you didn't just have it check if it's greater than (or equal) to `target_size` here and increase `size`?

Reply:
If I just allow increasing size by factor of 2, then we would obtain not enough results in a case where filter removes more then half of the results.